packages/grid/x-data-grid/src/hooks/features/filter/useGridFilter.tsx

```diff
@@ -78,10 +79,24 @@ export const useGridFilter = (
   const applyFilters = React.useCallback<GridFilterApi['unstable_applyFilters']>(() => {
     apiRef.current.setState((state) => {
       const filterModel = gridFilterModelSelector(state, apiRef.current.instanceId);
-      const isRowMatchingFilters =
+      const isRowMatchingStandardFilters =
```

Comment:
I propose to rename `buildAggregatedFilterApplier` => `buildAggregatedFilterItemsApplier` and to create a new `buildAggregatedFilterApplier`:

```ts
const buildAggregatedFilterApplier = (filterModel, apiRef) => {
  const isRowMatchingFilterItems = buildAggregatedFilterItemsApplier(filterMode, apiRef);
  const isRowMatchingQuickFilter = buildAggregationQuickFilterApplier(filterModel, apiRef);

  if (isRowMatchingFilterItems == null && isRowMatchingQuickFilter == null) {
    return null
  }

  if (isRowMatchingFilterItems == null) {
    return isRowMatchingQuickFilter
  }

  if (isRowMatchingQuickFilter == null) {
    return isRowMatchingFilterItems
  }
  
  return (rowId, shouldApplyFilter) => isRowMatchingFilterItems(rowId, shouldApplyFilter) && isRowMatchingQuickFilter(rowId, shouldApplyFilter)
}
```

(with the correct params typing of course)

That way, in `useGridFilter` you just have the same code as before.
I find it more readable.